corgidrp/detector.py

```diff
@@ -520,7 +528,149 @@ def calc_sat_fwc(emgain_arr,fwcpp_arr,fwcem_arr,sat_thresh):
+    """combine the dataset of residual image frames of neptune or uranus and create flat field 
+    	and associated error
+
+    	Args:
+        	residual_images (np.array): residual image frames of neptune and uranus 
+        	cent (np.array): centroid of the image frames
+        	planet (str):   name of the planet neptune or uranus
+        	band (str):  band of the observation band1 or band4
+        	
+    	Returns:
+        	full_qe (np.array): ideal flat field
+        	full_residuals (np.array): flat field from uranus or neptune images
+        	percens_resel (np.array): percentage error of the flat field
+    """
+    
+    n = 420
+    full_qe=np.random.normal(1,.03,(n,n))
+    full_residuals = np.zeros((n,n))
+    rad1 = 1.26; rad4 = 1.75
+    aperture1 = CircularAperture((np.ceil(rad1), np.ceil(rad1)), r=rad1)
+    aperture4 = CircularAperture((np.ceil(rad4), np.ceil(rad4)), r=rad4)
+    mask1= aperture1.to_mask().data
+    mask4= aperture4.to_mask().data
+    if planet=='neptune':
+        rad = 50
+    elif planet=='uranus':
+        rad=65
+
+    for i in np.arange(len(residual_images)):
+        nx = np.arange(0,residual_images[i].shape[1])
+        ny = np.arange(0,residual_images[i].shape[0])
+        nxx,nyy = np.meshgrid(nx,ny)
+        nrr = np.sqrt((nxx-rad-5)**2 + (nyy-rad-5)**2)
+        nrr_copy = nrr.copy()
+        nrr_copy[nrr<rad] = residual_images[i][nrr<rad] 
+        nrr_copy[nrr>=rad] = None 
+        ymin = int(cent[i][0])
+        ymax = int(cent[i][1])
+        xmin = int(cent[i][2])
+        xmax = int(cent[i][3])
+
+        bool_innotzero = np.logical_and(nrr<rad,full_residuals[ymin:ymax,xmin:xmax]!=0)
+        bool_iniszero = np.logical_and(nrr<rad,full_residuals[ymin:ymax,xmin:xmax]==0)
+        bool_outisnotzero = np.logical_and(nrr>=rad,full_residuals[ymin:ymax,xmin:xmax]!=0)
+        
+        full_residuals[ymin:ymax,xmin:xmax][bool_innotzero] = (nrr_copy[bool_innotzero] + full_residuals[ymin:ymax,xmin:xmax][bool_innotzero])/2
+        full_residuals[ymin:ymax,xmin:xmax][bool_iniszero] = nrr_copy[bool_iniszero]
+        full_residuals[ymin:ymax,xmin:xmax][bool_outisnotzero] = full_residuals[ymin:ymax,xmin:xmax][bool_outisnotzero]
+
+    if band == '4':
+        full_qe_resel = ndimage.convolve(full_qe, mask4)
+        full_residuals_resel = ndimage.convolve(full_residuals,mask4)
+        percens_resel = (((full_qe_resel - full_residuals_resel)/full_qe_resel)*100)
+    
+    elif band== '1':
+        full_qe_resel = ndimage.convolve(full_qe, mask1)
+        full_residuals_resel = ndimage.convolve(full_residuals,mask1)
+        percens_resel = (((full_qe_resel - full_residuals_resel)/full_qe_resel)*100)
+    
+    full_residuals[full_residuals==0] = None
+    nx = np.arange(0,percens_resel.shape[1])
+    ny = np.arange(0,percens_resel.shape[0])
+    nxx,nyy = np.meshgrid(ny,nx)
+    nrr = np.sqrt((nxx-n/2)**2 + (nyy-n/2)**2)
+    full_residuals[nrr>165]= 1
```

Comment:
More constants here (165 and 302). Do we always expect these to be hard coded?

Reply:
The 165 pixels come from the radius of the coronagraph FOV for which the flat field is created. Since the corgidrp flatfield is 1024x1024 pixels, I padded 302 pixels on either side of this FOV with 1s. Would you prefer these to be nan's? 

<img width="637" alt="Screenshot 2024-08-14 at 10 21 10 AM" src="https://github.com/user-attachments/assets/5447dcdc-4521-4a89-af93-63e3cdf5aacc">

Reply:
1s is fine. I think the main thing is possibly allowing these to be optional keywords that we can change in the future

Reply:
okay! I will make all the hard-coded values as keywords